main.py

```diff
@@ -413,24 +396,57 @@ def handle_app_mention_events(body, logger, say):
             thread_chats[thread_id] = []
         message_history = thread_chats[thread_id]
     else:
-        message_history = []
+        thread_chats[event_id] = []
+        message_history = thread_chats[event_id]
 
+    print(f"YYY {pre_prompt}")
     result = agent.run_sync(pre_prompt + user_prompt, message_history=message_history)
     message_history.extend(result.new_messages())
+    return result
 
-    say(text=result.output, thread_ts=thread_id or event["ts"])
-    slack_app.client.reactions_remove(channel=channel, name='hourglass_flowing_sand', timestamp=event["ts"])
 
 
-# Convert '<@USERID> msg' to 'msg'
+# Convert '<@USERID> msg' to 'msg' (Slack) or '@user:matrix.org msg' to 'msg' (Matrix)
 def strip_userid(msg: str):
     elems = msg.split(' ', 1)
-    if not elems[0].startswith('<@'):
+    if not elems[0].startswith('<@') and not elems[0].startswith('@'):
         return msg.strip()
     if len(elems) > 1:
         return elems[1].strip()
     return ''
 
 
 if __name__ == "__main__":
-    SocketModeHandler(slack_app, os.environ["SLACK_APP_TOKEN"]).start()
+    nest_asyncio.apply() # Apply nest_asyncio to allow nested event loops
```

Comment:
Do we need this though in Slack mode?

Reply:
not 100% sure. I could move it down to the `else` so it only applies to Matrix mode if you like.